Utility/Utilities.cs

```diff
@@ -40,30 +42,35 @@ public partial class Utilities {
             }
 
             Logger.Debug($"Request URL: {request.Method} {request.RequestUri}");
-            if (request.Method != HttpMethod.Get && request.Method != HttpMethod.Head) {
-                Logger.Trace($"Request body:{Environment.NewLine}{request.Content?.ReadAsStringAsync(ct).Result}");
+            if (request.Method != HttpMethod.Get
+                    && request.Method != HttpMethod.Delete
+                    && request.Method != HttpMethod.Trace
+                    && request.Method != HttpMethod.Options
+                    && request.Method != HttpMethod.Head) {
+                var requestContent = await (request.Content?.ReadAsStringAsync(ct) ?? Task.FromResult(string.Empty));
+                Logger.Trace($"Request body:{Environment.NewLine}{requestContent}");
             }
 
-            var client = new HttpClient();
+            using var client = new HttpClient();
             HttpResponseMessage response = null;
             int i = 1;
 
-            try {
-                response = await client.SendAsync(request, ct);
-            } catch (WebException ex) {
-                while (i < 4) {
-                    await Task.Delay(TimeSpan.FromSeconds(1), ct);
-                    Logger.Error($"HTTP request to {request.RequestUri} failed: {ex.Message}. Retry attempt {i}");
-
+            while (i < 4) {
+                try {
                     response = await client.SendAsync(request, ct);
+                    break;
+                } catch (WebException ex) {
+                    Logger.Error($"HTTP request to {request.RequestUri} failed: {ex.Message}. Retry attempt {i}");
```

Comment:
The retry logic wasn't working when the second attempt would throw an error. rearranging it should fix it.